pkgs/servers/nosql/eventstore/default.nix

```diff
@@ -1,72 +1,79 @@
-    dotnet-sdk
-    mono
-    Nuget
-  ];
-
-  # that dependency seems to not be required for building, but pulls in libcurl which fails to be located.
-  # see: https://github.com/EventStore/EventStore/issues/1897
-  patchPhase = ''
-    for f in $(find . -iname "*.csproj"); do
-      sed -i '/Include="Microsoft.SourceLink.GitHub"/d' $f
-    done
-  '';
+  # Fixes application reporting 0.0.0.0 as its version.
+  MINVERVERSIONOVERRIDE = version;
 
-  buildPhase = ''
-    export FrameworkPathOverride=${mono}/lib/mono/4.7.1-api
+  dotnet-sdk = dotnetCorePackages.sdk_5_0;
+  dotnet-runtime = dotnetCorePackages.aspnetcore_5_0;
+
+  nativeBuildInputs = [ git glibcLocales bintools ];
+
+  runtimeDeps = [ mono ];
 
-    # disable default-source so nuget does not try to download from online-repo
-    nuget sources Disable -Name "nuget.org"
-    # add all dependencies to a source called 'nixos'
-    for package in ${toString deps}; do
-      nuget add $package -Source nixos
+  nugetBinariesToPatch = if stdenv.isLinux then [
+    "grpc.tools/2.39.1/tools/linux_x64/protoc"
+    "grpc.tools/2.39.1/tools/linux_x64/grpc_csharp_plugin"
+  ] else [ ];
+
+  postConfigure = ''
+    # Fixes execution of native protoc binaries during build
+    for binary in $nugetBinariesToPatch; do
+      path="$HOME/.nuget/packages/$binary"
+      echo "Patching $path"
+      ${patchelf}/bin/patchelf --set-interpreter "$(cat $NIX_BINTOOLS/nix-support/dynamic-linker)" $path
     done
 
-    dotnet restore --source nixos src/EventStore.sln
-    dotnet build --no-restore -c Release src/EventStore.sln
+    # Fixes git execution by GitInfo on mac os
+    substituteInPlace "$HOME/.nuget/packages/gitinfo/2.0.26/build/GitInfo.targets" \
+      --replace "<GitExe Condition=\"Exists('/usr/bin/git')\">/usr/bin/git</GitExe>" " " \
+      --replace "<GitExe Condition=\"Exists('/usr/local/bin/git')\">/usr/local/bin/git</GitExe>" ""
   '';
 
-  installPhase = ''
-    mkdir -p $out/{bin,lib/eventstore}
-    cp -r bin/Release/* $out/lib/eventstore
-    makeWrapper "${mono}/bin/mono" $out/bin/eventstored \
-      --add-flags "$out/lib/eventstore/EventStore.ClusterNode/net471/EventStore.ClusterNode.exe"
-  '';
+  nugetDeps = ./deps.nix;
+
+  projectFile = "src/EventStore.ClusterNode/EventStore.ClusterNode.csproj";
 
   doCheck = true;
+  testProjectFile = "src/EventStore.Projections.Core.Tests/EventStore.Projections.Core.Tests.csproj";
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/EventStore.ClusterNode --version
+
+    temp="$(mktemp -d)";
+    $out/bin/EventStore.ClusterNode --insecure \
+      --db "$temp/data" \
+      --index "$temp/index" \
+      --log "$temp/log" \
+      -runprojections all --startstandardprojections \
+      --EnableAtomPubOverHttp &
 
-  checkPhase = ''
-    dotnet test src/EventStore.Projections.Core.Tests/EventStore.Projections.Core.Tests.csproj -- RunConfiguration.TargetPlatform=x64
+    PID=$!
+
+    sleep 30s;
+    kill "$PID";
```

Comment:
Why do we sleep for 30 seconds here? Is eventstore a server software that daemonises?

Reply:
Yes, theres no way to run it in a way that executes and closes. This script is a little wonky I admit, but I think it's the best way that actually tests execution of this program